ApiDoctor.Validation/OData/Parameter.cs

```diff
@@ -44,11 +44,21 @@ public Parameter()
         [XmlAttribute("Type"), ContainsType]
         public string Type { get; set; }
 
+        private bool _isNullable;
+
         [XmlAttribute("Nullable"), MergePolicy(MergePolicy.Ignore)]
-        public bool ValueOfNullableProp { get; set; }
+        public bool ValueOfNullableProp
+        {
+            get { return _isNullable; }
+            set
+            {
+                _isNullable = value;
+                ValueOfNullablePropSpecified = true;
+            }
+        }
 
         [XmlIgnore]
-        public bool ValueOfNullablePropSpecified => this.ValueOfNullableProp;
+        public bool ValueOfNullablePropSpecified { get; set; }
```

Comment:
curious why this was needed. what i did before was make it default to not outputting the attribute if it's set to its default value

Reply:
Nullable is true if not specified per spec. If Nullable is set to false in the input csdl, then ValueOfNullablePropSpecified was false and the Nullable attribute was not serialized to the output metadata.

Reply:
Omitting attributes wasn't the problem. Typically, teams only add attributes when the values are not the defaults. The problem was that the Nullable attribute was being set to a non-default value and the attribute wasn't being written to the output metadata file.